forte/data/data_store.py

```diff
@@ -966,6 +966,83 @@ def _is_annotation_tid(self, tid: int) -> bool:
         else:
             raise KeyError(f"Entry with tid {tid} not found.")
 
+    def _create_new_entry(
+        self, type_name: str, attribute_data: List, tid: Optional[int] = None
+    ) -> List:
+        r"""This function generates a new entry with default fields.
+        The new entry is in the format used to be stored in Data Stores.
+
+        Args:
+            type_name: The fully qualified type name of the new entry.
+            attribute_data: It is a list that stores attributes relevant to
+                the entry being added. In order to keep the number of attributes
+                same for all entries, the list is populated with trailing None's.
```

Comment:
When did the population happen?

Reply:
It happens during function call. An example is - [https://github.com/Pushkar-Bhuse/forte/blob/7a62e1899ed476a3c24f8f60bb29842b5c69eb29/forte/data/entry_converter.py#L112](url)